### DISCUSSION TITLE
Configure UI per task

### DISCUSSION THREAD

**Discussion by cAttte** [vercel/turborepo#9167] [Ideas]
### Goals

the `ui` field in `turbo.json` applies to all turbo tasks, but certain types of tasks are more approriate with either UI. this feature would allow the user to specify `ui` for specific tasks.

for example, a `dev` task is usually long-running and even interactive, so it's esential that it uses the `tui`. but a more streamlined task like `format` will typically output very few (if any) lines of text, so having all of its outputs aggregated in the `stream` way is much more convenient than replacing the entire screen with a faux tabbed terminal that will just close itself after a few seconds.

### Proposal

each task object could simply have a `ui` key, which would override the identical root `ui` key.

**Comment by anthonyshew** [comment]
Sounds like a similar discussion as here: https://github.com/vercel/turborepo/discussions/9091

I'm wondering what behavior you'd expect to see when running something like `turbo run dev format`, and you've assigned different `ui` values to different tasks.

**Comment by anthonyshew** [comment]
Closing as this ask doesn't seem to make much sense. There can only be one UI but many tasks, so there's no way to establish a true precedence.
